docs/contributors/github.rst

```diff
@@ -0,0 +1,42 @@
+################
+Github etiquette
+################
+
+How we work together.
+
+Where to learn
+--------------
+
+If you are new to ``git`` or Github, here are some resources to get you started:
+
+- `Git tutorial <https://www.atlassian.com/git>`_
+- `Github help <https://help.github.com/>`_
+- `Github tutorial <https://docs.github.com/en/get-started/quickstart/hello-world>`_
+- `How to contribute to Open Source <https://opensource.guide/how-to-contribute/>`_
+- `A successful branching mode <https://nvie.com/posts/a-successful-git-branching-model/>`_
+
+Typical workflow
+----------------
+
+1. Create a branch with a meaningful name like ``fix-xyz-victor`` or ``feature-abc-victor``.
+2. Commit to this branch as often as you want.
+    1. Commit messages should be meaningful and concise.
+    2. Try to commit in a granular way, not too many changes at once.
+3. As soon as you have a couple commits, push your branch to Github.
+4. As soon as you have pushed, create a **draft** pull request.
```

Comment:
Why is it good etiquette to create a draft PR? Just to let know others of it? If that's really good etiquette, combined with frequent pushing, then I definitely need to change my email notification settings haha

Reply:
It all depends on the kind of project, I think in a research group of various async collaborators it is important to be able to refer to bits of code easily (i.e. have a pushed branch instead of sending screenshots or partial bits of code) but also for other to grasp easily what you're working on and what changes you are planning.

All of this is to be put in perspective of the cost: very low (a couple buttons).

And yes, you should definitely update your email preferences for the current repo!

Reply:
Apparently you cannot disable email notifications about just Draft PRs: https://github.com/orgs/community/discussions/10409